contracts/actions/AaveDeposit.sol

```diff
@@ -38,8 +38,8 @@ contract AaveDeposit is IAaveDeposit {
 
         uint256 balanceBefore = aToken.scaledBalanceOf(address(this));
 
-        if (IERC20(token).allowance(address(this), address(lendingPool)) < amount) {
-            IERC20(token).approve(address(lendingPool), amount);
+        if (IERC20(token).allowance(address(this), address(lendingPool)) == 0) {
```

Comment:
And if by chance this allowance isn't 0? Please add an else statement (or use approve(0) -> approve(amount) pattern)